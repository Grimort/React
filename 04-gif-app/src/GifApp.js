import { useState } from "react";
import { AddCategory } from "../src/components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifApp = ({ defaultCategories = ["Memes"] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  // const handleAdd = () => {
  //   return setCategories([...categories, "cat4"]);
  // };

  return (
    <>
      <h2>GifApp</h2>
      <hr />
      <AddCategory setCategories={setCategories}></AddCategory>
      {/* <button onClick={handleAdd}>Add</button> */}
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category}></GifGrid>
        ))}
      </ol>
    </>
  );
};

export default GifApp;
