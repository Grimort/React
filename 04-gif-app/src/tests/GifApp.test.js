import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifApp } from "../GifApp";

describe("Testing GifApp component", () => {
  test("should to show the component correctly", () => {
    const wrapper = shallow(<GifApp />);
    expect(wrapper).toMatchSnapshot();
  });
});
