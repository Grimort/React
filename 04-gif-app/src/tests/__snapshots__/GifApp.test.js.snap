// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifApp component should to show the component correctly 1`] = `
<Fragment>
  <h2>
    GifApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Memes"
      key="Memes"
    />
  </ol>
</Fragment>
`;
